### PR TITLE
Upgrade @rjsf v5 dependencies to beta.18

### PR DESCRIPTION
**Pull request by dagda1**
Signed-off-by: Paul Cowan <paul.cowan@cutting.scot>

## Hey, I just made a Pull Request!

It would be good to test the recent `beta.17` changes in the `@rjsf/-v5` dependencies.

This PR also moves back to using the `@rjsf/validator-ajv8` validator.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
@dagda - there's beta.18 now, do you want to move to that instead? Also, not sure if we're going to get a release before they go stable so you might have to test it with the nightlies but does that work ok for you?

**Comment by dagda1** [comment]
> @dagda - there's beta.18 now. Do you want to move to that instead? Also, not sure if we're going to get a release before they go stable so you might have to test it with the nightlies but does that work ok for you?

Good catch. I did not check for any upgrades.  Nightlies are perfect for testing.

I've upgraded, and `formData` is now optional, so please check my logic for adding an initializer in the `Stepper` `onSubmit`.

File: plugins/scaffolder/package.json [hunk 1]
```diff
@@ -56,9 +56,9 @@
     "@material-ui/lab": "4.0.0-alpha.57",
     "@react-hookz/web": "^20.0.0",
     "@rjsf/core": "^3.2.1",
-    "@rjsf/core-v5": "npm:@rjsf/core@5.0.0-beta.16",
```
Comment (benjdlambert) [inline plugins/scaffolder/package.json:59]: I actually think that these deps can be removed now as I think `FormProps` is exported from the `scaffolder-react` package entirely.

**Review by benjdlambert — APPROVED** [review]
Cheers @dagda1.

**Pull request merged**